skfda/exploratory/stats/_functional_transformers.py

```diff
@@ -75,3 +75,136 @@ def local_averages(
         for i in range(n_intervals)
     ]
     return np.asarray(integrated_data)
+
+
+def _calculate_curves_occupation_(
+    curve_y_coordinates: np.ndarray,
+    curve_x_coordinates: np.ndarray,
+    interval: Tuple,
+) -> np.ndarray:
+    y1, y2 = interval
+
+    # Reshape original curves so they have one dimension less
+    new_shape = curve_y_coordinates.shape[1::-1]
+    curve_y_coordinates = curve_y_coordinates.reshape(
+        new_shape[::-1],
+    )
+
+    # Calculate interval sizes on the X axis
+    x_rotated = np.roll(curve_x_coordinates, 1)
```

Comment:
Why rotated?

Reply:
This piece of code obtains the size of each interval. If you take a grid; for example [1, 3, 6, 9], you rotate it one position to the left: [3, 6, 9, 1] and then you subtract both arrays [3, 6, 9, 1]  - [1, 3, 6, 9] as a result you obtain the size of each interval : [2, 3, 3, -8]. We have to take out the last number as it does not count. Interval (1, 3) has size 2, (3, 6) size 3 and (6,9) size 3. -> [2, 3, 3]